### PR TITLE
Fix Kotlin integration in the Gradle plugin

### LINKED ISSUE #11867: Gradle build fails after upgrade to 1.7.2
After upgrade to 1.7.2, all maven projects work just fine, but gradle projects fail with:
```
Step #0 - "Build and Push Image": > Task :quarkusGenerateCode
Step #0 - "Build and Push Image": preparing quarkus application
Step #0 - "Build and Push Image": kotlin scripting plugin: root project 'my-service'.compileQuarkus-generated-sourcesKotlin - configuration not found: quarkus-generated-sourcesKotlinScriptDef, the plugin is probably applied by a mistake
Step #0 - "Build and Push Image": 
Step #0 - "Build and Push Image": > Task :quarkusGenerateCodeTests FAILED
Step #0 - "Build and Push Image": preparing quarkus application
Step #0 - "Build and Push Image": kotlin scripting plugin: root project 'my-service'.compileQuarkus-test-generated-sourcesKotlin - configuration not found: quarkus-test-generated-sourcesKotlinScriptDef, the plugin is probably applied by a mistake
Step #0 - "Build and Push Image": 
Step #0 - "Build and Push Image": FAILURE: Build failed with an exception.
Step #0 - "Build and Push Image": 
Step #0 - "Build and Push Image": * What went wrong:
Step #0 - "Build and Push Image": Execution failed for task ':quarkusGenerateCodeTests'.
Step #0 - "Build and Push Image": > Failed to generate sources in the QuarkusPrepare task
```

Any thoughts about why it happened and how to fix that?

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #11867

### CODE REVIEW

File: devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java [hunk 1]
```diff
@@ -139,13 +139,9 @@ public void execute(Task test) {
...
-                    Task compileKotlinTask = tasks.findByName("compileKotlin");
-                    // TODO: proper support of kotlin
-                    if (compileKotlinTask == null) {
-                        JavaCompile compileTestJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_TEST_JAVA_TASK_NAME);
-                        compileTestJavaTask.dependsOn(quarkusGenerateCodeTests);
-                        quarkusGenerateCodeTests.setSourceRegistrar(compileTestJavaTask::source);
-                    }
+                    JavaCompile compileTestJavaTask = (JavaCompile) tasks.getByName(JavaPlugin.COMPILE_TEST_JAVA_TASK_NAME);
+                    compileTestJavaTask.dependsOn(quarkusGenerateCodeTests);
+                    quarkusGenerateCodeTests.setSourceRegistrar(compileTestJavaTask::source);
```
Comment (glefloch) [inline devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java:144]: maybe we should do the same in the kotlin plugin block but I don't know if can add the `kotlin.jvm` plugin as a project dependency to get the type of `compileKoltin` task.

**Review by glefloch — APPROVED** [review]
Looks good to me.

**Review by aloubyansky — APPROVED** [review]

**Comment by gsmet** [comment]
If this is a regression, I think we need a test. Don't know how hard it will be to add one though.

File: devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java (hunk 1 above)
Comment (gastaldi) [reply devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPlugin.java:144]: Yeah, not sure if adding a hard dependency on the kotlin plugin is a good idea in this case. We could revisit that after this issue is resolved

**Comment by gastaldi** [comment]
@gsmet yeah, sorry. I updated my PR with an integration test. 

Can we "re-release" the 1.7.2.Final Gradle plugin in the plugin portal with this fix?

**Pull request merged**